lib/features/caching/config/cache_client.dart

```diff
@@ -0,0 +1,29 @@
+
+import 'package:hive/hive.dart';
+
+abstract class CacheClient<T> {
+
+  String get tableName;
+
+  Future<Box<T>> openTable();
```

Comment:
it is specific for `Box`, so I propose the class name `HiveCache<T>`

Reply:
or you should create a super class `Cache<T>` without this method, then you create `HiveCache<T>` and add this method